metrics/web/docs/features/firestore_emulator_for_driver_tests/01_firestore_emulator_for_driver_tests.md

```diff
@@ -0,0 +1,232 @@
+2. The web application uses the Firestore instance configured to use the emulator while running integration tests.
+### Prototyping
+
+> Create a simple prototype to confirm that implementing this feature is possible.
+
+To run the driver tests under the Firebase emulator we should follow the next steps:
+
+1.  Run the Firebase emulator before running the app under tests.
+2.  Configure the application under tests to use the Firebase emulator instance before running tests.
+3.  Run the driver tests as usual.
+
+Before we run the emulator, we need to check the configurations in the `firebase.json` file.
+
+The emulator will take Security Rules configuration from the `firestore` configuration key,
+so make sure it links to the right file location.
+
+```json
+    "firestore": {
+        "rules": "firestore/rules/firestore.rules",
+        ...
+    }
+```
+
+The default emulator's port is 8080.
+To customize that you can specify an additional key in the `firebase.json` file.
+
+```json
+    "emulators": {
+        "firestore": {
+            "port": "8080"
+        },
+        ...
+    }
+```
+
+The emulator erases the data inside the Firestore each time it starts.
+
+To prepare the test data, we can manually create necessary collections and documents through the emulator's UI and save it using the following command:
+
+```bash
+firebase emulators:export export_directory
+```
+
+The specified `export_directory` will be created if it does not already exist. If the specified directory exists, you will be prompted to confirm that the previous export data should be overwritten. For more information on using the command consider [the following link](https://firebase.google.com/docs/emulator-suite/install_and_configure#export_and_import_emulator_data).
+
+Now we can run the emulator with the initial data by adding the following flag `--import=import_directory`, where the `import_directory` is the directory that holds the exported data.
+
+So with that in place, we can run the emulator by using the following command:
+
+```bash
+    firebase emulators:start --only firestore --import=import_directory
+```
+
+Now the emulator is running and it is time to use it in the app's integration tests.
+
+In the `setUpAll` method of the tests we need to configure the Firestore instance to use the emulator:
+
+```dart
+setupAll(() {
+    Firestore.instance.settings(
+        host: 'localhost:8080',
+        sslEnabled: false,
+        persistenceEnabled: false,
+    );
+});
+```
+
+The specified port in the `host` argument must be equal to the emulator's port.
+
+# Design
+
+### Architecture
+
+> Fundamental structures of the feature and context (diagram).
+
+![Firestore emulator class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/firestore_emulator_design/metrics/web/docs/features/firestore_emulator_for_driver_tests/diagrams/firestore_emulator_class_diagram.puml)
```

Comment:
Would it be better to move this diagram to be closer to the section where we are explaining the process of configuring new arguments and passing them to the Flutter driver tests?